API Raises an error when a negative threshold is chosen in VarianceThreshold (#20207)
scikit-learn/scikit-learn: doc/whats_new/v1.0.rst

  :pr:`17169` by :user:`Dmytro Lituiev <DSLituiev>`
  and :user:`Julien Jerphanion <jjerphan>`.

> - |Fix| Fixed a bug in class:`feature_selection.VarianceThreshold`
>   when a variance threshold is negative.

Reviewer: To properly link the class and a minor adjustment to the text:

```suggestion
- |API| Raises an error in :class:`feature_selection.VarianceThreshold`
  when the variance threshold is negative.
```

Looking at this again, I think this is more of an API change where we are performing more validation.
